concent_api/utils/helpers.py

```diff
@@ -166,15 +169,27 @@ def get_storage_source_file_path(subtask_id, task_id):
     return get_storage_file_path('source', subtask_id, task_id)
 
 
-def calculate_maximum_download_time(size: int) -> int:
+def calculate_maximum_download_time(size: int, rate: int = constants.DEFAULT_UPLOAD_RATE) -> int:
```

Comment:
Please add assertions for invalid values. Does the function return anything sensible for negative `size` or `rate`? Does it work with negative `DOWNLOAD_LEADIN_TIME`? If not, this should be guarded with `assert`. Otherwise you have to test and support these parameter combinations (which in my opinion is a waste of time).

Reply:
:+1: